bindings/pyroot/pythonizations/test/rdf_filter.py

```diff
@@ -0,0 +1,123 @@
+
+        remaining_cols = ["Long64_t", "ULong64_t", "Long_t", "ULong_t"]
+        for col_name in remaining_cols:
+            rdf = ROOT.RDataFrame(col_name,"./TestData_" + col_name + ".root")
+            df = pd.DataFrame(rdf.AsNumpy())
+            filter = "filter_" + TYPE_TO_SYMBOL[col_name] # string of filter name
+            func = eval(filter); # function of the filter
+            x = df[col_name].mean(); x = float(x);
+            filtered = rdf.Filter(func, extra_args={'x':x})
+            res_root = filtered.AsNumpy()[col_name]
+            filtered2 = rdf.Filter(f"{col_name} > {x}")
+            res_root2 = filtered2.AsNumpy()[col_name]
+            self.assertTrue(np.array_equal(res_root,res_root2))
+        os.remove("./TestData.root")
+        for col_name in remaining_cols:
+            os.remove("./TestData_" + col_name + ".root")
+    
+    def test_filter_overload1_a(self):
+        """
+        Test to verify the first overload (1.a) of filter
+        Filter(callable, col_list, name)
+        """
+        rdf = ROOT.RDataFrame(5).Define("x", "(double) rdfentry_")
+        def x_greater_than_2(x):
+            return x>2
+        fil1 = rdf.Filter(x_greater_than_2, ["x"], "x is more than 2")
+        self.assertTrue(np.array_equal(fil1.AsNumpy()["x"], np.array([3, 4])))
+
+    def test_filter_overload1_b(self):
+        """
+        Test to verify the first overload (1.b) of filter
+        Filter(callable, col_list)
+        """
+        rdf = ROOT.RDataFrame(5).Define("x", "(double) rdfentry_")
+        def x_greater_than_2(x):
+            return x>2
+        fil1 = rdf.Filter(x_greater_than_2, ["x"])
+        self.assertTrue(np.array_equal(fil1.AsNumpy()["x"], np.array([3, 4])))
+    
+    def test_filter_overload1_c(self):
+        """
+        Test to verify the first overload (1.c) of filter
+        Filter(callable)
+        """
+        rdf = ROOT.RDataFrame(5).Define("x", "(double) rdfentry_")
+        def x_greater_than_2(x):
+            return x>2
+        fil1 = rdf.Filter(x_greater_than_2)
+        self.assertTrue(np.array_equal(fil1.AsNumpy()["x"], np.array([3, 4])))
+    
+    def test_filter_overload3(self):
+        """
+        Test to verify the third overload of filter
+        Filter(callable, na,e)
+        """
+        rdf = ROOT.RDataFrame(5).Define("x", "(double) rdfentry_")
+        def x_greater_than_2(x):
+            return x>2
+        fil1 = rdf.Filter(x_greater_than_2, "x is greater than 2")
+        self.assertTrue(np.array_equal(fil1.AsNumpy()["x"], np.array([3, 4])))
+    
+    def test_filter_overload_4_a(self):
+        """
+        Test to verify the fourth (4.a) overload of filter
+        Filter(expression, name)
+        """
+        rdf = ROOT.RDataFrame(5).Define("x", "(double) rdfentry_")
+        fil1 = rdf.Filter(lambda x: x>2, "x is greater than 2")
+        self.assertTrue(np.array_equal(fil1.AsNumpy()["x"], np.array([3, 4])))
+
+    def test_filter_overload_4_b(self):
```

Comment:
I don't think we need to test the overload of Filter that accepts an expression with C++ code. This is extensively tested everywhere. More interesting would be to test a Filter that accepts cppyy wrappers, but it's not in the scope of this PR